### ISSUE TITLE
Wrong decrypt result 

### ISSUE THREAD

**Issue by QuangAnh84** [delletenebre/dart_des#5]
Hi guys, 

I can decrypt the message but it always has some strange characters at the end such as  

I have function like this:

```
Future<String> decryptTDES(String data, String password) async {
      List<int> iv = [0, 0, 0, 0, 0, 0, 0, 0];
      List<int> decrypted;
      DES3 des3CBC = DES3(key: md5.convert(utf8.encode(password)).bytes, mode: DESMode.CBC, iv: iv);
      decrypted = des3CBC.decrypt(base64.decode(data));
      return utf8.decode(decrypted);
  }
```
And it also takes so long time (~30-45s) to decrypt a message 1200 characters (base64 format)
Can someone help me?

**Comment by farhantandia** [comment]
I can agree for the decrypt time, it takes so long to finish, this algo not suitable for decrypting big message (e.g >100kb file).

**Comment by Lua12138** [comment]
> I can agree for the decrypt time, it takes so long to finish, this algo not suitable for decrypting big message (e.g >100kb file).

Hello @farhantandia , 

Do you have any suggestions for the content of about 300KB?  (3DES ECB PKCS5)

**Comment by farhantandia** [comment]
@Lua12138 I prefer to use AES

**Comment by Lua12138** [comment]
I know DES is not secure.

But I have no choice. The third-party service I need to use requires me to
use DES on messages.

M. Farhan Tandia ***@***.***> 于2023年7月5日周三 19:36写道：

**Comment by farhantandia** [comment]
@Lua12138 I think just in this package the decryption method is not optimized yet, thus the processing time is so long to complete, because Ive compare it with .net c# base DES algorithm, and the processing time is way faster than this package.

**Comment by Lua12138** [comment]
Yes, it uses dynamically allocated memory during the operation, resulting
in the need to frequently copy and create new memory areas. This leads to
an exponential decrease in speed for larger messages.

tripledes_nullsafety | Flutter Package (pub.dev)
<https://pub.dev/packages/tripledes_nullsafety>

This component can quickly encrypt and decrypt the DES algorithm

M. Farhan Tandia ***@***.***> 于2023年7月6日周四 20:27写道：
